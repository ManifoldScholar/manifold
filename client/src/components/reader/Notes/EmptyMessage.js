import React, { PureComponent } from "react";
import PropTypes from "prop-types";

export default class EmptyMessage extends PureComponent {
  static displayName = "Notes.EmptyMessage";

  static propTypes = {
    annotated: PropTypes.bool
  };

  template(title, body) {
    return (
      <div className="notes-message">
        <h5 className="heading-primary">
          {title}
        </h5>
        <p>
          {body}
        </p>
      </div>
    );
  }

  renderNoResults() {
    const title = "No annotations found";
    const body = `None of your annotations match the filters set above. Adjust the filters
    to show more results.`;
    return this.template(title, body);
  }

  renderNotYetAnnotated() {
    const title = "You haven't annotated this text yet";
    const body = `Once you've highlighted or annotated a passage in this text, it will
          appear here. To annotate or highlight, select a passage from the text
          and click the appropriate button on the pop-up menu.`;
    return this.template(title, body);
  }

  render() {
    return this.props.annotated
      ? this.renderNoResults()
      : this.renderNotYetAnnotated();
  }
}
