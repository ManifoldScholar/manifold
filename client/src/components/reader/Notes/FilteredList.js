import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { Notes } from "components/reader";

export default class FilteredList extends PureComponent {
  static displayName = "Notes.FilteredList";

  static propTypes = {
    section: PropTypes.object,
    sortedAnnotations: PropTypes.array,
    handleSeeAllClick: PropTypes.func,
    handleFilterChange: PropTypes.func,
    handleVisitAnnotation: PropTypes.func,
    currentSectionId: PropTypes.string,
    filter: PropTypes.object,
    annotated: PropTypes.bool,
    loaded: PropTypes.bool
  };

  static defaultProps = {
    annotations: []
  };

  renderHeading() {
    return (
      <div className="drawer-bar">
        <h2 className="drawer-title" onClick={this.props.handleSeeAllClick}>
          Your Notes
        </h2>
        {this.props.sortedAnnotations.length > 0
          ? <button
              onClick={this.props.handleSeeAllClick}
              className="button-primary"
            >
              See all
            </button>
          : null}
      </div>
    );
  }

  renderList() {
    if (this.props.sortedAnnotations.length === 0) {
      return <Notes.EmptyMessage annotated={this.props.annotated} />;
    }
    return (
      <ul>
        {this.props.sortedAnnotations.map(group => {
          return (
            <Notes.Partial.Group
              key={group.sectionId}
              annotations={group.annotations}
              sectionName={group.name}
              readerSection={this.props.section}
              visitHandler={this.props.handleVisitAnnotation}
            />
          );
        })}
      </ul>
    );
  }

  render() {
    return (
      <div>
        {this.renderHeading()}
        <Notes.Partial.Filters
          filterChangeHandler={this.props.handleFilterChange}
          filter={this.props.filter}
        />
        <nav>
          {this.props.loaded ?
          this.renderList()
          : null }
        </nav>
      </div>
    );
  }
}
