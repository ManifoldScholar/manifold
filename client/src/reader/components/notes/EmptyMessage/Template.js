import React from "react";
import PropTypes from "prop-types";

function Template({ title, body }) {
  return (
    <div className="notes-message">
      <h3 className="heading-primary">{title}</h3>
      <p>{body}</p>
    </div>
  );
}

Template.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired
};

export default Template;
