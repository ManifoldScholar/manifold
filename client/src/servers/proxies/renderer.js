import config from "config";
import proxy from "http-proxy-middleware";
import ch from "../../helpers/consoleHelpers";
import createStore from "../../store/createStore";
import exceptionRenderer from "../../helpers/exceptionRenderer";
import manifoldBootstrap from "bootstrap";
import CookieHelper from "helpers/cookie/Server";

const ssrRenderUrl = `http://${config.services.client.domain}:${config.services.client.sparePort}`;

export default function makeRendererProxy(requestHandler) {
  const reactServerProxy = proxy({
    target: ssrRenderUrl,
    changeOrigin: true,
    logLevel: "silent",

    onError: (err, req, res) => {
      ch.error(`Error proxying to port ${ssrRenderUrl}`);
      ch.error(`Perhaps the SSR service is in the process of reloading?`);
      ch.error(`Falling back to SSR rescue service only`);

      const store = createStore();

      const render = () => {
        try {
          requestHandler(req, res);
        } catch (error) {
          if (error.code === "MODULE_NOT_FOUND") {
            const msg =
              "Waiting for initial Webpack build to complete. Wait a few seconds " +
              "and reload.";
            ch.error(msg);
            res.setHeader("Content-Type", "text/html");
            res.end(msg);
          } else {
            ch.error(
              `Rendering fallback failed to render inn server-developmet.js`
            );
            res.setHeader("Content-Type", "text/html");
            res.end(exceptionRenderer(error));
          }
        }
      };

      const promises = [];
      const cookie = new CookieHelper(req, res);
      promises.push(manifoldBootstrap(store.getState, store.dispatch, cookie));
      Promise.all(promises).then(render, render);
    }
  });
  return reactServerProxy;
}
