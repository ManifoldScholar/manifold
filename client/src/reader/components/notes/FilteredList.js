import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import Partial from "./partial";
import EmptyMessage from "./EmptyMessage";

export default class FilteredList extends PureComponent {
  static displayName = "Notes.FilteredList";

  static propTypes = {
    section: PropTypes.object,
    sortedAnnotations: PropTypes.array,
    handleSeeAllClick: PropTypes.func,
    handleFilterChange: PropTypes.func,
    handleVisitAnnotation: PropTypes.func,
    currentSectionId: PropTypes.string,
    filter: PropTypes.object,
    annotated: PropTypes.bool,
    loaded: PropTypes.bool
  };

  static defaultProps = {
    annotated: false,
    loaded: false
  };

  static defaultProps = {
    annotations: []
  };

  get hasAnnotations() {
    return this.props.sortedAnnotations.length > 0;
  }

  renderHeading() {
    const { handleSeeAllClick, handleFilterChange, filter } = this.props;
    return (
      <div className="notes-filtered-list__header">
        <Partial.Filters
          filterChangeHandler={handleFilterChange}
          filter={filter}
        />
        <button
          onClick={handleSeeAllClick}
          className="notes-filtered-list__see-all button-primary"
        >
          <span className="button-primary__text">See all</span>
        </button>
      </div>
    );
  }

  renderList() {
    if (!this.hasAnnotations) {
      return <EmptyMessage annotated={this.props.annotated} />;
    }
    return (
      <ul className="notes-filtered-list__section-list">
        {this.props.sortedAnnotations.map(group => {
          return (
            <Partial.Group
              key={group.sectionId}
              annotations={group.annotations}
              sectionName={group.name}
              readerSection={this.props.section}
              visitHandler={this.props.handleVisitAnnotation}
            />
          );
        })}
      </ul>
    );
  }

  render() {
    return (
      <div className="notes-filtered-list">
        {this.renderHeading()}
        <div>{this.props.loaded ? this.renderList() : null}</div>
      </div>
    );
  }
}
