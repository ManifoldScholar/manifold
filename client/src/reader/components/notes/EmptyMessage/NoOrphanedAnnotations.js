import React from "react";
import Template from "./Template";
import { useTranslation } from "react-i18next";

const NoOrphanedAnnotations = () => {
  const { t } = useTranslation();

  return (
    <Template
      title={t("reader.menus.notes.no_annotations")}
      body={t("reader.menus.notes.no_orphans_message")}
    />
  );
};

export default NoOrphanedAnnotations;
