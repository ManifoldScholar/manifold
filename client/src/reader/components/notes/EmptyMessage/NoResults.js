import React from "react";
import Template from "./Template";
import { useTranslation } from "react-i18next";

function NoResults() {
  const { t } = useTranslation();
  return (
    <Template
      title={t("reader.menus.notes.no_annotations")}
      body={t("reader.menus.notes.no_matches_with_filters")}
    />
  );
}

export default NoResults;
