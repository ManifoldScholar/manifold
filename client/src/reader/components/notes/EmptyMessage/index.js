import NoResults from "./NoResults";
import TextNotAnnotatedByGroup from "./TextNotAnnotatedByGroup";
import TextNotAnnotatedByMe from "./TextNotAnnotatedByMe";
import NoOrphanedAnnotations from "./NoOrphanedAnnotations";

export default {
  NoResults,
  TextNotAnnotatedByGroup,
  TextNotAnnotatedByMe,
  NoOrphanedAnnotations
};
