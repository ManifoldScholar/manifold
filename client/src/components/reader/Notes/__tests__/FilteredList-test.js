import React from "react";
import renderer from "react-test-renderer";
import FilteredList from "../FilteredList";
import build from "test/fixtures/build";
import { shallow } from "enzyme/build/index";
import EmptyMessage from "../EmptyMessage";

describe("Reader.Notes.FilteredList Component", () => {
  const sortedAnnotations = [
    {
      name: "Test",
      sectionId: 1,
      annotations: [build.entity.annotation("1"), build.entity.annotation("2")]
    }
  ];

  const section = build.entity.textSection("3");

  const filter = {
    formats: ["highlight", "annotation", "bookmark"]
  };

  const clickMock = jest.fn();

  const component = renderer.create(
    <FilteredList
      sortedAnnotations={sortedAnnotations}
      handleSeeAllClick={clickMock}
      handleFilterChange={clickMock}
      section={section}
      filter={filter}
      loaded
    />
  );

  it("renders correctly", () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("doesn't render to null", () => {
    let tree = component.toJSON();
    expect(tree).not.toBe(null);
  });

  it("renders an empty message when there are no annotations", () => {
    const wrapper = shallow(
      <FilteredList
        sortedAnnotations={[]}
        handleSeeAllClick={clickMock}
        handleFilterChange={clickMock}
        section={section}
        filter={filter}
	    loaded
      />
    );
    expect(wrapper.find(EmptyMessage).length).toBe(1);
  });
});
