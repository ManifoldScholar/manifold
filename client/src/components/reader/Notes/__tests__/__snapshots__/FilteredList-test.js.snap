// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reader.Notes.FilteredList Component renders correctly 1`] = `
<div>
  <div
    className="drawer-bar"
  >
    <h2
      className="drawer-title"
    >
      <div
        onClick={[MockFunction]}
        role="button"
      >
        Your Notes
      </div>
    </h2>
    <button
      className="button-primary"
      onClick={[MockFunction]}
    >
      See all
    </button>
  </div>
  <nav
    className="filters"
  >
    <label
      className="label"
    >
      Show your:
    </label>
    <div
      className="checkbox-group"
    >
      <label
        className="checkbox"
        htmlFor="Highlights-checkbox-1"
      >
        <input
          checked={true}
          id="Highlights-checkbox-1"
          onChange={[Function]}
          type="checkbox"
        />
        <div
          aria-hidden="true"
          className="control-indicator"
        >
          <i
            className="manicon manicon-check"
          />
        </div>
        Highlights
      </label>
      <label
        className="checkbox"
        htmlFor="Annotations-checkbox-2"
      >
        <input
          checked={true}
          id="Annotations-checkbox-2"
          onChange={[Function]}
          type="checkbox"
        />
        <div
          aria-hidden="true"
          className="control-indicator"
        >
          <i
            className="manicon manicon-check"
          />
        </div>
        Annotations
      </label>
    </div>
  </nav>
  <nav>
    <ul>
      <li>
        <div
          className="item"
          onClick={[Function]}
          role="button"
        >
          <i
            aria-hidden="true"
            className="manicon manicon-caret-down"
          />
          <label>
            Test
          </label>
          <span
            className="screen-reader-text"
          >
            Expand Notes
          </span>
        </div>
        <ul
          className=""
        >
          <li
            className="item"
            onClick={[Function]}
          >
            <span
              className="screen-reader-text"
            />
            <i
              aria-hidden="true"
              className="manicon manicon-null"
            />
            <span>
              Gods, Earths, and 85ers
            </span>
          </li>
          <li
            className="item"
            onClick={[Function]}
          >
            <span
              className="screen-reader-text"
            />
            <i
              aria-hidden="true"
              className="manicon manicon-null"
            />
            <span>
              Gods, Earths, and 85ers
            </span>
          </li>
        </ul>
      </li>
    </ul>
  </nav>
</div>
`;
